### DISCUSSION TITLE
Can it update images of swarm stack ?

### DISCUSSION THREAD

**Discussion by ZealousMacwan** [containrrr/watchtower#695] [Q&A] [answered]
Can I use watchtower to update images of stack running in swarm mode ?

**Comment by simonaronsson** [comment]
Maybe! We've had some users reporting some success using swarm with watchtower previously. I don't use swarm myself, and to my knowledge, nor does @piksel, so support is definitely on a might-work-who-knows-basis.

**Reply by piksel** [reply]
No, never used swarm at all. @ZealousMacwan please let us know if you try it!

**Comment by piksel** [comment, marked as answer]
No, it does not, since docker swarm pins images to the current hash and therefore will always have the "latest version", since no other image could have the same hash.

Try using [shepherd](https://github.com/containrrr/shepherd) instead. It was purposely made for updating swarms and doesn't need to do all the shenanigans that watchtower does to make it work with regular docker containers.
